### PR TITLE
Use Self where possible, some minor refactoring

### PR DESCRIPTION
**Pull request by adamnemecek**

### CODE REVIEW

**Comment by Narsil** [comment]
@adamnemecek thanks for this. This seems a nice little lint. Does any software detect those ?

**Comment by Narsil** [comment]
Ahaa the test seem to point that it wasn't software but a full search+replace. Was it ?

**Comment by adamnemecek** [comment]
I'm not aware. It was done with a text editor. I have pushed an updated version that fixed the failing test. However there were some failing tests even before I made my changes.

**Comment by Narsil** [comment]
It seems to be in the nursery, so it'll come sooner or later to `clippy`: https://rust-lang.github.io/rust-clippy/master/#use_self

Your editor probably just includes those too.

Thanks for the fixes !

File: tokenizers/src/pre_tokenizers/unicode_scripts/scripts.rs [hunk 1]
```diff
@@ -144,1934 +144,1940 @@ pub enum Script {
...
-        0x17000..=0x187EC => Script::Tangut,
-        0x18800..=0x18AF2 => Script::Tangut,
-        _ => Script::Any,
+impl Script {
+    pub fn new(c: char) -> Self {
+        match c as u32 {
+            0x0000..=0x001F => Self::Common,
+            0x0020 => Self::Common,
+            0x0021..=0x0023 => Self::Common,
+            0x0024 => Self::Common,
```
Comment (Narsil) [inline tokenizers/src/pre_tokenizers/unicode_scripts/scripts.rs]: This is an arguably good change, but it does NOT belong in this PR.

File: tokenizers/src/tokenizer/encoding.rs [hunk 2]
```diff
@@ -110,11 +110,7 @@ impl Encoding {
 
     /// Return the number of sequences combined in this Encoding
     pub fn n_sequences(&self) -> usize {
-        if self.sequence_ranges.is_empty() {
-            1
-        } else {
-            self.sequence_ranges.len()
-        }
+        self.sequence_ranges.len().max(1)
```
Comment (Narsil) [inline tokenizers/src/tokenizer/encoding.rs:113]: Same, I would rather not include this in this PR.

The `Whatever` -> `Self` is a pretty big change that touches most of the code base.
We should keep everything that might have other side effects OUT so that it's easier to find the problem later.

On this particular one, I am not particularly fond of the final result, `std::cmp::max(1, self.sequences_ranges.len())` feels much more natural (should we make a change).

**Review by Narsil — COMMENTED** [review]
Thank you for those changes.

Do you mind creating another PR for the two seemingly unrelated changes?

I think they're ok but I would prefer to not mix those changes if possible (I can do the split if you want).

File: tokenizers/src/tokenizer/encoding.rs (hunk 2 above)
Comment (adamnemecek) [reply tokenizers/src/tokenizer/encoding.rs]: Done.

**Review by Narsil — APPROVED** [review]
LGTM
